test/functionalTest/cases/3001_mqtt/subscribe_and_update.test

```diff
@@ -0,0 +1,129 @@
+# Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+MQTT Notifications
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0-255 IPV4 -mqttHost localhost
+
+--SHELL--
+echo "01. POST /v2/entities, creating E1/T1"
+echo "====================================="
+payload='{
+  "id": "E1",
+  "type": "T1",
+  "A1": {
+    "value": 23,
+    "type": "Float"
+  },
+  "A2": {
+    "value": 720,
+    "type": "Integer"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+echo "02. Subscribe to E1 on condition A2"
+echo "==================================="
+payload='{
+  "description": "Test subscription",
+  "subject": {
+    "entities": [
+      {
+        "id": "E1",
+        "type": "T1"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "A2"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "mqtt://mqtttest.com"
+    },
+    "attrs": [
+      "A1"
+    ]
+  },
+  "expires": "2040-01-01T14:00:00.00Z",
+  "throttling": 1
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+echo "03. Update entity attribute A1 to trigger notification"
```

Comment:
Ideally we should have an `accumulator-mqtt.py` program similar to the one we already have now for HTTP (accumulator-server.py). I mean, a subscriber of the MQTT broker able to listen & store MQTT-based notification from Orion and able to print the stored notifications in standard output when the proper command (an special MQTT topic) is received.

Have you proficiency in Python and the related MATT libraries ([paho-mqtt](https://pypi.org/project/paho-mqtt/))? :) In positive case, you could develop this script. In negative case I could try to have a look and contribute with it to the feature/mqtt_notifications_poc (however, in this case we should find a way it doesn't block the ongoing work, as I'd do it in a best effort way and it could take some time).

(I'm suggesting Python as it is the one we typically use in our testing tooling and it is better not to diverge using a different lenguage)

Reply:
Actually, the accumulator doesn't really speak HTTP, it just saves whatever text it received, and then spits it out on request.
I know absolutely nothing about MQTT ...
If it's a text based protocol, like HTTP, then the accumulator would work as is.
If not (which I hope), you'd need some python lib to add mqtt functionality (either inside the accumulator or write a new accumulator)

Reply:
Current accumulator is not going to work... The library we use there (Flask) is for HTTP and doesn't work for MQTT, which needs another libraries (the paho-mqtt I mention above). It is not rocket science, but it would required some knowledgue in Python to implement it. I can do it (best effort :) if @McMutton cannot.

I'd suggest do it in a separated .py file, for simplicity and cleaness.

Reply:
@fgalan I have Python experience, I'll give it a try. In any case, get ready for more questions :)

> I'd suggest do it in a separated .py file, for simplicity and cleaness. 

Sure :+1: